### ISSUE TITLE
3.11.0 worker restart regression

### ISSUE THREAD

**Issue by mkrogh** [concourse/concourse#2162]
# Bug Report

* Concourse version: 3.11.0
* Deployment type (BOSH/Docker/binary): docker
* Infrastructure/IaaS: Ubuntu 16.04 VM
* Did this used to work? yes in 3.10.0

When restarting a worker (or rebooting the machine), 3.11.0 workers fail to start if the `volumes.img` already exists.
Removing the file allows the worker to start, but fails on next restart/reboot.

**Comment by mkrogh** [comment]
```
{"timestamp":"1523969394.548230410","source":"baggageclaim","message":"baggageclaim.fs.run-command.failed","log_level":2,"data":{"args":["bash","-e","-x","-c","\n\t\tif [ ! -e $IMAGE_PATH ] || [ \"$(stat --printf=\"%s\" $IMAGE_PATH)\" != \"$SIZE_IN_BYTES\" ]; ...
{"timestamp":"1523969394.548875570","source":"baggageclaim","message":"baggageclaim.failed-to-set-up-driver","log_level":2,"data":{"error":"failed to create btrfs filesystem: exit status 1"}}failed to create btrfs filesystem: exit status 1
Main process exited, code=exited, status=1/FAILURE
```

My current workaround it to just run `rm` on the `volumes.img` before starting the workers, since the db migrations makes it so you cannot run the older version of the `web` command.

**Comment by AXington** [comment]
I think this is a bug in ubuntu or docker, not in concourse. I had the same issue (at least same log errors) on 3.10.0 this morning. Mine was volumes mounted through overlayfs instead of volumes.img, but it was the same baggage claim errors. The solution for me was to umount the overlayfs, reboot, and I was fine.

**Comment by billimek** [comment]
We've seen this type of error in our installation too.  Running in a similar setup:

* Deployment type (BOSH/Docker/binary): docker
* Infrastructure/IaaS: Ubuntu 16.04 VM

We sometimes see this when someone needs to make an update to the `docker-compose.yml` and/or destroy and re-create the worker for whatever reason.  When they start up the new worker, there are problems with the worker not starting properly or error about the `volumes.img` file.  It doesn't seem like a very clean solution to have to delete this file first before starting a new worker.

**Comment by xylphid** [comment]
Got the same error each time the worker restart with this configuration : 

* Concourse version : 3.13.0
* Deployment type (BOSH/Docker/binary): docker
* Infrastructure:IaaS : Mint 18.3

**Comment by fewknow** [comment]
+1 .  I upgraded from 3.9.1 to 3.14.1 
After this upgrade everything turned orange and had this error. 
```
volume '9f7d378a-fd70-4c0e-412c-36c56364a478' disappeared from worker 'a1380cc4-c7f1-4ad1-84ab-67af113428f9'
```
<img width="1072" alt="screen shot 2018-07-01 at 10 16 40 pm" src="https://user-images.githubusercontent.com/22297806/42142316-a876d202-7d7c-11e8-8ab3-84a29d761e23.png">

<img width="649" alt="screen shot 2018-07-01 at 10 15 57 pm" src="https://user-images.githubusercontent.com/22297806/42142315-a82adff0-7d7c-11e8-91ae-c9ed76cae295.png">

I currently run 10 workers.   After recreating them with 
```
bosh -e mgmt recreate -d concourse worker/a0a5b764-9205-4794-97f8-08d1b0ad4b3f
```
The error went away and everything seems to be fine.

**Comment by billimek** [comment]
Bumping this issue:

When running via docker-compose, if the worker container crashes and restarts (say due to a OOM error) - or even during any type of restart - we frequently see something like,

```
{"timestamp":"1533239376.210015059","source":"baggageclaim","message":"baggageclaim.fs.run-command.failed","log_level":2,"data":{"args":["bash","-e","-x","-c","\n\t\tif [ ! -e $IMAGE_PATH ] || [ \"$(stat --printf=\"%s\" $IMAGE_PATH)\" != \"$SIZE_IN_BYTES\" ]; then\n\t\t\ttouch $IMAGE_PATH\n\t\t\ttruncate -s ${SIZE_IN_BYTES} $IMAGE_PATH\n\t\tfi\n\n\t\tlo=\"$(losetup -j $IMAGE_PATH | cut -d':' -f1)\"\n\t\tif [ -z \"$lo\" ]; then\n\t\t\tlo=\"$(losetup -f --show $IMAGE_PATH)\"\n\t\tfi\n\n\t\tif ! file $IMAGE_PATH | grep BTRFS; then\n\t\t\t/opt/concourse/4.0.0/assets/btrfs/mkfs.btrfs --nodiscard $IMAGE_PATH\n\t\tfi\n\n\t\tmkdir -p $MOUNT_PATH\n\n\t\tif ! mountpoint -q $MOUNT_PATH; then\n\t\t\tmount -t btrfs $lo $MOUNT_PATH\n\t\tfi\n\t"],"command":"/bin/bash","env":["PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin","MOUNT_PATH=/opt/concourse/volumes","IMAGE_PATH=/opt/concourse/volumes.img","SIZE_IN_BYTES=94813585408"],"error":"exit status 1","session":"3.1","stderr":"+ '[' '!' -e /opt/concourse/volumes.img ']'\n++ stat --printf=%s /opt/concourse/volumes.img\n+ '[' 94813585408 '!=' 94813585408 ']'\n++ losetup -j /opt/concourse/volumes.img\n++ cut -d: -f1\n+ lo=/dev/loop2\n+ '[' -z /dev/loop2 ']'\n+ file /opt/concourse/volumes.img\nbash: line 11: file: command not found\n+ grep BTRFS\n+ /opt/concourse/4.0.0/assets/btrfs/mkfs.btrfs --nodiscard /opt/concourse/volumes.img\n/opt/concourse/volumes.img appears to contain an existing filesystem (btrfs).\nERROR: use the -f option to force overwrite of /opt/concourse/volumes.img\n","stdout":"btrfs-progs v4.15\nSee http://btrfs.wiki.kernel.org for more information.\n\n"}}
```

The important part from this seems to be:
**volumes.img appears to contain an existing filesystem (btrfs)**

... which prevents the worker from ever starting.  Requiring the user to manually clean-up this file from the host filesystem in order to get a worker to start.

Basically, once you create a worker, it looks like you cannot restart the worker without first manually removing the sparse-image file.

Is there any way to bake-in this cleanup as part of concourse itself?

**Comment by billimek** [comment]
I think this error is actually going to be fixed by https://github.com/concourse/concourse-docker/pull/15

Removing the `volumes.img` was probably just masking the check using the missing `file` command.  Can't wait to see this PR included in a release!

**Comment by OvermindDL1** [comment]
Still happening in 4.2.1, server reboot causes `failed to create btrfs filesystem: exit status 1` to be logged and then the worker dies.

**Comment by UncleSamSwiss** [comment]
@billimek unfortunately it doesn't seem to work.

We have the same problem with Concourse 5.0 and the following versions:

`# docker -v`
`Docker version 18.06.1-ce, build e68fc7a`
`# docker-compose -v`
`docker-compose version 1.23.2, build 1110ad01`
`# uname -a`
`Linux seazoet3 4.15.0-46-generic #49-Ubuntu SMP Wed Feb 6 09:33:07 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`

**Comment by UncleSamSwiss** [comment]
For everybody ending up here with the same error I reported, the **workaround** is:

1. Run `docker-compose down`
2. Remove the volume(s) related to the "concourse" container (for us these were two)
3. Run `docker-compose up -d`
4. Run `fly -t myproject workers` and prune all workers that are `stalled`

**Comment by punalpatel** [comment]
Thank you @UncleSamSwiss These steps worked for me.

Run 

docker-compose down; docker system prune --volumes; docker-compose up -d

**Comment by vito** [comment]
Pretty sure this was fixed by https://github.com/concourse/concourse-docker/pull/15 but then regressed with 5.0+ - it should be fixed again in v5.1 which we hope to have out this week or next week. :+1:

**Comment by pothos** [comment]
Had the same problem with v5.3.0.
https://github.com/concourse/concourse/issues/2162#issuecomment-478299961 solved it.

**Comment by acarsercan** [comment]
I'm running v5.3.0 and still having this problem with concourse running inside kubernetes

```volume '9fdd49f5-333e-4838-74b5-27b5d858cb80' disappeared from worker 'concourse-worker-3'```

**Comment by Suvankar123** [comment]
I am facing same issue with concourse helm chart with image version 7.7.I had deployed previously with baggageclaim driver set to : naive.But it did not work since all went down.Again I redployed the total helm via brtfs & making mkfs.brtfs flag as enabled setting the required binary,but still facing the same issue
